src/libraries/Common/src/Interop/Unix/System.Security.Cryptography.Native/Interop.Ssl.cs

```diff
@@ -325,6 +327,33 @@ internal enum SslErrorCode
     }
 }
 
+namespace System.Runtime.InteropServices.Marshalling
+{
+    [CustomMarshaller(typeof(SafeX509Handle), MarshalMode.ManagedToUnmanagedOut, typeof(NullableX509HandleMarshaller.Out))]
+    internal static class NullableX509HandleMarshaller
+    {
+        public struct Out
+        {
+            private SafeX509Handle? _handle;
+
+            public void FromUnmanaged(IntPtr value)
+            {
+                if (value != IntPtr.Zero)
+                {
+                    _handle = new SafeX509Handle();
```

Comment:
`FromUnmanaged` method is not supposed to throw exceptions. This allocates and so it can throw exceptions.

This should be something like this to follow the marshaller contract. @jkoritzinsky Thoughs?
```csharp
public struct Out
{
    private IntPtr _handle;
    
    public void FromUnmanaged(IntPtr value)
    {
        _handle = value;
    }
    
    public SafeX509Handle? ToManaged()
    {
        if (_handle == IntPtr.Zero)
            return null;
            
        var safeHandle = new SafeX509Handle(); 
             
        Marshal.InitHandle(safeHandle, _handle);
        _handle = IntPtr.Zero; // Ownership transferred
             
        return safeHandle;       
    }
    
    public void Free()
    {
        if (_handle != IntPtr.Zero)
            Interop.Crypto.X509Destroy(_handle);
    }
 }
```

Reply:
Yes, I agree that your design better fits the model.

An even better option would be to pre-allocate the handle instance, if we’re trying to match the runtime and generator’s marshalling logic as closely as possible.

Reply:
> An even better option would be to pre-allocate the handle instance, if we’re trying to match the runtime and generator’s marshalling logic as closely as possible.

I believe that one of the motivations for this change is to avoid the safe handle allocation when the P/Invoke returns null, so the handle is not pre-allocated on purpose.

Reply:
That makes sense.

Reply:
It would be _nice_ if the standard functions supports nullability ... but I'm not sure if that is possible since the `Free` is type specific.